pkg/querier/cardinality/request.go

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package cardinality
```

Comment:
Placing this under `pkg/querier/cardinality` seems strange due to "querier" part. Perhaps `pkg/cardinality` would be a better fit, so that `querymiddleware` doesn't "reach out" to querier packages.

Reply:
Ok I can do it, but `querymiddleware` is already reaching `pkg/querier/...` subpackages.